subseas_flex_fcst/maproom.py

```diff
@@ -135,19 +185,30 @@ def local_plots(n_clicks, click_lat_lng, latitude, longitude):
             isnan = isnan + isnan_yt
         if isnan > 0:
             errorFig = pingrid.error_fig(error_msg="Data missing at this location")
-            return errorFig, errorFig, None, dlf.Marker(position=[lat, lng]), lat, lng 
+            return errorFig, errorFig, None, dlf.Marker(position=[lat, lng]), lat, lng
     except KeyError:
         errorFig = pingrid.error_fig(error_msg="Grid box out of data domain")
         return errorFig, errorFig, None, dlf.Marker(position=[lat, lng]), lat, lng
-    
-    # Get Issue date and Target season
-    # Hard coded for now as I am not sure how we are going to deal with time
-    issue_date = pd.to_datetime(["2022-04-01"]).strftime("%-d %b %Y").values[0]
-    target_start = pd.to_datetime(["2022-04-02"]).strftime("%-d %b %Y").values[0]
-    target_end = pd.to_datetime(["2022-04-08"]).strftime("%-d %b %Y").values[0]
 
+    # Get Issue date and Target season
+    issue_date_td = pd.to_datetime(fcst_var["T"].values)
+    issue_date = issue_date_td.strftime("%-d %b %Y")
+    #for leads they are currently set to be the difference in days fron target_start to issue date
+    if fcst_var["L"].values == "Week 1":
+        lead_time = 1
+        #lead_time = (issue_date_td + timedelta(days=(targetStart_add + 3))).strftime("%-d %b %Y") #from when I made lead time the midpoint date
+    elif fcst_var["L"].values == "Week 2":
+        lead_time = 8
+        #lead_time = (issue_date_td + timedelta(days=(targetStart_add +3))).strftime("%-d %b %Y")
+    elif fcst_var["L"].values == "Week 3":
+        lead_time = 15
+        #lead_time = (issue_date_td + timedelta(days=(targetStart_add+3))).strftime("%-d %b %Y")
+    elif fcst_var["L"].values == "Week 4":
+        lead_time = 22
```

Comment:
For this, is there a mathematical expression used to calculate the dates? Since I am failing to see how changing L to be integers like that would be any simpler in the way I have coded things.

Reply:
Okay. I am still deciding how to process on this part. If everything else is wrapped up, I think I will merge this PR and then start a new one to deal with all the lead time stuff, unless there are objections on that.